lib/requester/core.js

```diff
@@ -229,7 +229,9 @@ var dns = require('dns'),
             // returning error synchronously causes uncaught error because listeners are not attached to error events
             // on socket yet
             return setImmediate(function () {
-                callback(null, resolvedAddr, resolvedFamily);
+                // When options.all is set, the callback expects array of addresses instead of a single address
+                return options.all ? callback(null, [{ address: resolvedAddr, family: resolvedFamily }]) :
```

Comment:
We're just sending the available address and family as an array of object. Don't we have list of addresses?

Reply:
network.hostLookup option does not support multiple IPs at the moment. That is a separate feature improvement we can do after this PR maybe.